models/workitemtype_test.go

```diff
@@ -71,6 +74,8 @@ func TestMarshalEnumType(t *testing.T) {
 }
 
 func TestMarshalFieldDef(t *testing.T) {
+	resource.Require(t, resource.UnitTest)
```

Comment:
This is how `resource.Require()` is implemented right now:

```go
// Require checks if all the given environment variables ("envVars") are set
// and if one is not set it will skip the test ("t").
func Require(t *testing.T, envVars ...string) {
	for _, envVar := range envVars {
		if _, c := os.LookupEnv(envVar); c == false {
			t.Skipf(StSkipReason, envVar)
			return
		}
	}
}
```

So technically you can call `resource.require(t, "foo", "bar")` and the test `t` will be skipped if not all of the environment variables, `foo` and `bar`, are set.

If we define `resoure.None := ""` we have to add a check that checks the given environment variable for zero length before checking:

```go
// Require checks if all the given environment variables ("envVars") are set
// and if one is not set it will skip the test ("t").
func Require(t *testing.T, envVars ...string) {
	for _, envVar := range envVars {
		if len(envVar) > 0 {
			if _, c := os.LookupEnv(envVar); c == false {
				t.Skipf(StSkipReason, envVar)
				return
			}
		}
	}
}
```

This would work but when we run integrations tests, they will also run unit tests. Is that a wanted behavior? Do we see integration tests as a super-set of unit tests?

Or are you suggesting to just rename `resource.UnitTest` to `resource.None` and still define it as `ALMIGHTY_RESOURCE_UNIT_TEST`?